### ISSUE TITLE
Ref. "update a record in an elastic table"

### ISSUE THREAD

**Issue by AndrewElans** [MicrosoftDocs/powerapps-docs#5059]
Ref. [this issue](https://github.com/MicrosoftDocs/power-automate-docs/issues/1294#issue-2054936348).

Summary: updating one column in elastic table record with Power Automate causes data in all other columns to be wiped/erased.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 430deb7b-cad7-a516-222d-d3507b6912b9
* Version Independent ID: 430deb7b-cad7-a516-222d-d3507b6912b9
* Content: [Use elastic tables using code - Power Apps](https://learn.microsoft.com/en-us/power-apps/developer/data-platform/use-elastic-tables?tabs=sdk)
* Content Source: [powerapps-docs/developer/data-platform/use-elastic-tables.md](https://github.com/MicrosoftDocs/powerapps-docs/blob/main/powerapps-docs/developer/data-platform/use-elastic-tables.md)
* Service: **powerapps**
* Sub-service: **dataverse-developer**
* GitHub Login: @pnghub
* Microsoft Alias: **gned**

**Comment by bishalgoswami** [comment]
Thanks for your comment, @AndrewElans. This issue has been assigned to the doc author for review.

**Comment by AndrewElans** [comment]
The same bahaviour is true for web api operation

**Comment by JimDaly** [comment]
With Dataverse Web API both Upsert and Update use the `PATCH` HTTP method.
The difference is that Update must use the `If-Match: *` request header. This prevents an Upsert from occurring.
An Upsert doesn't use the `If-Match: *` request header.

With elastic tables, [Upsert operations will overwrite existing values](https://learn.microsoft.com/power-apps/developer/data-platform/use-elastic-tables?tabs=webapi#upsert-a-record-in-an-elastic-table).
In Power Automate, I believe the 'Update a row' action is using PATCH without the `If-Match: *` request header, which makes it an Upsert. 

I'll loop in the people to see how to best make this visible to people using Power Automate with elastic tables.

**Comment by AndrewElans** [comment]
Thanks, @JimDaly.

I've just tested the web api operations on the elastic table. Providing the result for reference, basically as explained in [this para](https://learn.microsoft.com/en-us/power-apps/developer/data-platform/use-upsert-insert-update-record?tabs=webapi#using-web-api).

- PATCH with header If-Match: * updates only the selected field of the existing entity.
- PATCH without header If-Match: * updates ALL fields of the existing entity, including "Created On". Fields that are not included in the Body are set to null.
- PUT updates only selected field

Power Automate flow though shall be fixed.

**Comment by JimDaly** [comment]
I've looped in the teams responsible for the documentation here:

- [Create and edit elastic tables - Power Apps | Microsoft Learn](https://learn.microsoft.com/power-apps/maker/data-platform/create-edit-elastic-tables)  
- [Use a flow to update a row in Dataverse - Power Automate | Microsoft Learn](https://learn.microsoft.com/power-automate/dataverse/update)

But I think this is a bug. Until it is fixed, I think the work around is to exactly what you shouldn't do with **standard** tables: 

- Retrieve all the columns
- Apply changes to that set, 
- Update to all columns

There isn't anything actionable for this article. It explains the API behavior correctly. Closing this.
